# Build Gwas Data Set
Objectif : using 1000 genomes and simulation process as [phenosim](append link), build gwas data set for qc, imputation, gwas and post analysis for [h3africa](append link) pipeline and script related for testing 
# Data set generation 
## General approach 
  * prepare data for simulation of phenotype and qc : 
    * Download file from 1000 Genomes project (release:20130502), transform data in plink format, change sex of participant in fam
  * Simulate phenotype using [h3agwas/assoc/simul-assoc.nf](append link) with [phenosim](), [plink]() ...
  * qc script to clean data (not in version 1) [h3agwas/qc/main.nf](append link)
  * imputed script to imputated data (not in version 1) [append link]()
  * gwas script for gwas  [h3agwas/assoc/main.nf](append link)
  * post gwas analysis [h3agwas/assoc/](append link)

## Version 2 
### Description 
  * exemple of data set of all chromosome and all individuals extract of 1000 genome
  * version 2 is in directory [V2](Append link) 
### Tutorial and preparation data
1. preparation data set

first step consist to generate data in plink format : time, download should be high due to high size 
need bash, plink, wgeti, awk, grep and R

```
cd V2/
bash extract_datatest.bash
```

also a nextflow version is available to create dataset


```
nextflow -c nf_extractdataset.nf run extract_datatest.config -profile slurm -resume
```

2. phenotype simulation (optional)

you can generate yourself a new data set, but one is already provide in `phenotype/simul/` a subsample of genotype is used, to simulate phenotype with 5 locus 

```
nextflow pull h3abionet/h3agwas
nextflow -c simul.config run h3abionet/h3agwas/assoc/simul-assoc.nf  --input_dir data --input_pat 1000G.h3aPos_0.1_0.05  --output_dir phenotype/  --out "pheno." -profile slurm -resume -r hackathon

```

3. qc 

We apply file with QC generated by previous step, we discarded sex because no X is in bedfile and used sex for case control checking, steps 
```
## create a phenotype file for case control
echo -e "FID\tIID\tPheno" > pheno
awk '{print $1"\t"$2"\t"$5}' data/1000G.h3aPos.fam  >> pheno
## launch qc 
nextflow run h3abionet/h3agwas/qc/main.nf  --input_dir data --input_pat 1000G.h3aPos --output_dir  qc  --case_control pheno --case_control_col Pheno --sexinfo_available false   -r hackathon -profile slurm -resume
```

4. imputation

TODO

5. GWAS



## Version 1 
* Example of data set of chromosome 1 (subsample of 1000G : position on chromosome 1 for dna cheap h3africa 
* version 1 is in directory [V1](Append link) 
* folder description 
 * `extract_1000GAndPosChr1.bash`  : bash script to generate data and simulate phenotype, output :
  * simul.config : config file of h3agwas pipeline for simulation :
   * simulate 3 data set, of 5 locus each with qtl strenght "0.1,0.3,0.1,0.2,0.2" and maf ? "0.05,0.000001,0.00000137,0.05,0.02"
  * `plkfile` folder : contains for chromosome 1 plink file of 1000 genomes and position of h3africa array 
    * `1000G.h3aPos.1.vcf.[bed/bim/fam]`: 2504 individuals with 163 842 positions of chromosomes 1
  * `Chr1Test/simul/` :
    * ̀ 1000G.h3aPos.1.vcf.ms.\_NumSimulation\_.pheno.pheno` : phenotype simulated for each individuals
    * `1000G.h3aPos.1.vcf.ms.\_NumSimulation\_.pheno.causal` : positions used to simulated phenotype with phenosim : 1000G.h3aPos.1.vcf.ms.\_NumSimulation\_.pheno.causal
* gwas : 
 * `gwas.config` : gwas config file for ha3gwas pipeline 
 * `gwas.bash` : bash to launch gwas config with slurm singularity

* limits :
  * doesn't work for boltlmm and LOCO, due to 1 chromosome
  * some issue with fastlmm and Insertion /Deletion 
  * QC pipeline have be not apply
  * Imputation pipeline have been not tested on data

### how to use 
clone branch

```
git clone jeantristanb/
```

data of chromosome is already generated, but if you want generated again

```
profile=slurm ## profile should be slurm, singularity
cd V1
rm -rf Chr1Test plkfile
bash extract_1000GAndPosChr1.bash $profile
```

do a gwas using singularity

```
nextflow -c launchgwas.config run h3abionet/h3agwas/assoc/main.nf -resume  -profile singularity -r hackathon
```

do a gwas using slurm and singularity
```
nextflow -c launchgwas.config run h3abionet/h3agwas/assoc/main.nf -resume  -profile slurmSingularity -r hackathon
```


# Data
* folder `data`
* ̀ integrated\_call\_samples_v2.20130502.ALL.ped` : information for 1000 genomes sample
* ̀ bim file for h3africa array`: to do

# Bin : 
* folder `bin`
* changesex.r : Rscript to change fam file and append sex
 

## General script 

## Requirements 
* shell/ bash /wget 
* [nextflow](Append Link) and pipeline :
  * [h3abionet/h3agwas]() branch hackathon at 9/08/19
  * [h3abionet/imputation]()
* [plink](plink)
* [R]() R-project
* [singularity](Append Link) or install software dependency of pipeline :
  * [h3abionet/h3agwas]() branch hackathon at 9/08/19
  * [h3abionet/imputation]()

## Contributing

Please read [CONTRIBUTING.md](https://gist.github.com/PurpleBooth/b24679402957c63ec426) for details on our code of conduct, and the process for submitting pull requests to us.

## Versioning

We use [SemVer](http://semver.org/) for versioning. For the versions available, see the [tags on this repository](https://github.com/your/project/tags).


## Authors

* **Brandenburg jean-tristan** 
* **Participant ha3bionet consortium, h3agwas, ha3imputation** 

## License

